txcache/accountBreadcrumb.go

```diff
@@ -28,15 +28,29 @@ func (breadcrumb *accountBreadcrumb) accumulateConsumedBalance(transferredValue
 	}
 }
 
-func (breadcrumb *accountBreadcrumb) updateLastNonce(lastNonce core.OptionalUint64) error {
+// updateFirstNonce is called only in the updateNonceRange method
+func (breadcrumb *accountBreadcrumb) updateFirstNonce(firstNonce core.OptionalUint64) {
+	breadcrumb.firstNonce = firstNonce
+}
+
+// updateNonceRange updates:
+// the last nonce of a sender breadcrumb.
+// the first nonce, in case the sender was previously ony a relayer.
```

Comment:
so if we first see a tx with a sender, then another tx with the same sender as relayer? should be ok, right?